### PR TITLE
PULL REQUEST: Story Mode Menu Scrolling

### PR DESCRIPTION
**Pull request by JVNpixels**
# Adds the ability to scroll in the StoryMenuState.
This pull request adds the ability for scrolling in the story mode menu, and it would be a good quality of life feature.

### CODE REVIEW

**Comment by JVNpixels** [comment]
I accidentally made 2 commits, but if this is good enough for merging, merge the Update StoryMenuState.hx one.

**Comment by amyspark-ng** [comment]
awesome

**Comment by JVNpixels** [comment]
Thank you! I'm not the best at GitHub but this is one step in the right direction.

**Comment by JVNpixels** [comment]
I wonder when these will get approved...

**Comment by Keoiki** [comment]
I think it'd be best if you made a commit in your own branch removing the duplicate StoryMenuState.hx file. I don't think repo admins can pick and choose which files to merge from a PR.

File: StoryMenuState.hx
Comment (gamerbross) [inline StoryMenuState.hx:1]: remove this duplicated file

**Comment by tposejank** [comment]
> I think it'd be best if you made a commit in your own branch removing the duplicate StoryMenuState.hx file. I don't think repo admins can pick and choose which files to merge from a PR.

They can change the base branch to another then merge there, afterwards cherry pick the commit to develop. Would be quite the hassle

File: source/funkin/ui/story/StoryMenuState.hx [hunk 1]
```diff
@@ -336,6 +336,16 @@ class StoryMenuState extends MusicBeatState
           changeDifficulty(0);
         }
 
+        if (FlxG.mouse.wheel < 0)
```
Comment (gamerbross) [inline source/funkin/ui/story/StoryMenuState.hx:339]: this wheel behaviour has things from html and non-html code, check the freeplay code as reference

**Review by gamerbross — CHANGES_REQUESTED** [review]

**Comment by JVNpixels** [comment]
The code should be fixed for HTML and Non HTML builds, and the duplicate file is now deleted.

**Comment by JVNpixels** [comment]
How are there 14 commits even though I made 4?

**Comment by moxie-coder** [comment]
> How are there 14 commits even though I made 4?

cuz the branch was changed

**Comment by gamerbross** [comment]
> How are there 14 commits even though I made 4?

the branch you made was forked from main and not develop
you should delete the branch, create it at put all the changes in a single commit (if possible) and make another pull request

**Review by gamerbross — CHANGES_REQUESTED** [review]
the branch you made was forked from main and not develop
you should delete the branch, create it from develop, put all the changes in a single commit (if possible) and make another pull request

File: source/funkin/ui/story/StoryMenuState.hx [hunk 2]
```diff
@@ -336,6 +336,24 @@ class StoryMenuState extends MusicBeatState
...
+        {
+          changeLevel(-Math.round(FlxG.mouse.wheel / 8));
+        }
+        else if (FlxG.mouse.wheel > 0)
+        {
+          changeLevel(-Math.round(FlxG.mouse.wheel / 8));
+        }
+        #end
+
+        // HTML and NON HTML builds mouse fix (JVN).
```
Comment (gamerbross) [inline source/funkin/ui/story/StoryMenuState.hx:355]: its not cool to try to sneek urself in the code lmao

Comment (tposejank) [reply source/funkin/ui/story/StoryMenuState.hx:355]: yep

**Comment by JVNpixels** [comment]
Alright, I have started a new PR here #2873
It should have all the things and changes you guys have requested me.

**Comment by JVNpixels** [comment]
Sorry for the inconveniences!

**Comment by tposejank** [comment]
Close this

**Pull request closed without merging**